src/qt/walletmodel.cpp

```diff
@@ -527,6 +527,7 @@ OperationResult WalletModel::PrepareShieldedTransaction(WalletModelTransaction*
                                                         bool fromTransparent,
                                                         const CCoinControl* coinControl)
 {
+    AssertLockHeld(pwalletMain->cs_wallet);
```

Comment:
Same as on `send.cpp`, this isn't being locked on the GUI. GUI side should never lock `cs_wallet` (and that is the future cleanup goal in the places where we do it, like in the txModel).